src/inputs/labeled-inputs.scss

```diff
@@ -28,169 +45,165 @@
+@mixin iui-input-label {
+  font-weight: $iui-font-weight-semibold;
+  grid-area: label;
+  align-self: center;
 
-      svg:not(.iui-icon) {
-        margin-right: 0;
+  &.iui-required {
+    &::after {
+      content: ' *';
+      @include themed {
+        color: t(iui-color-foreground-negative);
       }
     }
+  }
+}
 
-    > .iui-input,
-    > .iui-textarea {
-      padding-right: $iui-icons-default + $iui-l;
+/// Modifier on iui-input-label to place it inline.
+@mixin iui-input-label-inline {
+  margin: 0 $iui-m 0 0;
 
-      + .iui-message {
-        position: absolute;
-        top: 50%;
-        right: 0;
-        transform: translateY(-50%);
+  &.iui-required {
+    margin-right: $iui-xs * 1.5;
+  }
+}
 
-        svg:not(.iui-icon) {
-          margin-right: $iui-sm;
-        }
-      }
+/// Message shown below input
+@mixin iui-input-message {
+  font-size: $iui-font-size-small;
+  grid-area: message;
+  @include themed {
+    color: t(iui-text-color-muted);
+  }
 
-      &:last-child {
-        padding-right: $iui-sm;
-      }
-    }
+  a {
+    user-select: none;
   }
+}
 
-  // #region Statuses
-  &.iui-positive {
-    @include iui-inputs-status(
-      iui-color-foreground-positive,
-      iui-color-foreground-positive-overlay,
-      iui-color-foreground-positive-rgb
-    );
+/// Icon to be shown near an input.
+@mixin iui-input-icon {
+  display: flex;
+  grid-area: icon;
+  width: $iui-icons-default;
+  height: $iui-icons-default;
+  align-self: center;
+  @include themed {
+    fill: t(iui-icons-color);
   }
 
-  &.iui-warning {
-    @include iui-inputs-status(
-      iui-color-foreground-warning,
-      iui-color-foreground-warning-overlay,
-      iui-color-foreground-warning-rgb
-    );
+  &:not(:last-child) {
+    margin-right: $iui-xs;
   }
+}
 
-  &.iui-negative {
-    @include iui-inputs-status(
-      iui-color-foreground-negative,
-      iui-color-foreground-negative-overlay,
-      iui-color-foreground-negative-rgb
-    );
+/// Modifier on iui-input-icon to place it at the end of input.
+@mixin iui-input-icon-inline {
+  // stylelint-disable -- fix position in IE
+  -ms-grid-row: 3;
+  -ms-grid-column: 2;
+  -ms-grid-column-align: end;
+  // stylelint-enable
+
+  grid-area: inputs;
+  justify-self: end;
+  margin-right: $iui-sm;
+  position: relative;
+
+  &.iui-button {
+    height: 100%;
+    width: fit-content;
+    margin-right: 0;
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
   }
-  // #endregion
 }
 
-@mixin iui-inputs-status($foregroundColor, $foregroundOverlay, $foregroundRgb) {
+/// Mixin for applying status colors to input container.
+/// @arg status - must be one of: positive, negative, warning
+/// @arg iconSelector - selector to apply status fill on. Defaults to .iui-input-icon
+/// @arg textSelector - selector to apply text color on. Defaults to .iui-message
+@mixin iui-input-status($status, $iconSelector: '.iui-input-icon', $textSelector: '.iui-message') {
```

Comment:
for now, I dont see this being reused, not sure we need `iconSelector` and `textSelector` args.